build/common.go

```diff
@@ -0,0 +1,281 @@
+package build
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+var exname string = ""
+
+func getExecutableName(os string, arch string) string {
+	if exname == "" {
+		var err error
+		exname, err = getExecutableFromPluginJSON()
+		if err != nil {
+			exname = "set_exe_name_in_plugin_json" // warning in the final name?
+		}
+	}
+
+	exeName := fmt.Sprintf("%s_%s_%s", exname, os, arch)
+	if "windows" == os {
+		exeName = fmt.Sprintf("%s.exe", exeName)
+	}
+	return exeName
+}
+
+func getExecutableFromPluginJSON() (string, error) {
+	byteValue, err := readFileBytes(path.Join("src", "plugin.json"))
+	if err != nil {
+		return "", err
+	}
+
+	var result map[string]interface{}
+	err = json.Unmarshal([]byte(byteValue), &result)
+	if err != nil {
+		return "", err
+	}
+	return result["executable"].(string), nil
+}
+
+func readFileBytes(file string) ([]byte, error) {
+	jsonFile, err := os.Open(file)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = jsonFile.Close()
+	}()
+
+	return ioutil.ReadAll(jsonFile)
+}
+
+func findRunningPIDs(exe string) []int {
+	pids := []int{}
+	out, err := sh.Output("pgrep", exe[:15]) // full name does not match, only the prefix (on linux anyway)
```

Comment:
Guessing this doesn't work on Windows?

Reply:
no... but lets tackle that bridge later :)